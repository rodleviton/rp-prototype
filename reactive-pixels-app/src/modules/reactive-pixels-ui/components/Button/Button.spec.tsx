import * as React from "react";
import { cleanup, render } from "react-testing-library";
import WithTheme from "../../utils/test/WithTheme";
import { Button } from "./";

const mockOnClickFn = jest.fn().mockImplementation(() => console.log("Hello"));

function renderComponent(props: any) {
  const { variant = "default" } = props;
  const { size = "small" } = props;

  return render(
    <WithTheme>
      <Button data-testid="button-under-test" {...props}>
        I am a {size} {variant} button
      </Button>
    </WithTheme>
  );
}

describe("<Button />", () => {
  beforeEach(() => {
    jest.restoreAllMocks();
  });

  afterEach(cleanup);

  describe("variants", () => {
    it("should render a default <Button /> component", () => {
      const { container } = renderComponent({
        onClick: mockOnClickFn,
        variant: "default"
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it("should render a `primary` <Button /> component", () => {
      const { container } = renderComponent({
        onClick: mockOnClickFn,
        variant: "primary"
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it("should render a `secondary` <Button /> component", () => {
      const { container } = renderComponent({
        onClick: mockOnClickFn,
        variant: "secondary"
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  describe("sizes", () => {
    it("should render a `small` <Button /> component", () => {
      const { container } = renderComponent({
        onClick: mockOnClickFn,
        size: "small"
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it("should render a `small` <Button /> component", () => {
      const { container } = renderComponent({
        onClick: mockOnClickFn,
        size: "large"
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
