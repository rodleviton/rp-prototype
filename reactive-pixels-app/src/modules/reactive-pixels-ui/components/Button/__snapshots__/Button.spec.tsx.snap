// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> sizes should render a \`small\` <Button /> component 1`] = `
<button
  class="css-pmc161 css-2f61t6 css-1dhsa7o css-9fbhz8"
  data-testid="button-under-test"
>
  I am a 
  small
   
  default
   button
</button>
`;

exports[`<Button /> sizes should render a \`small\` <Button /> component 2`] = `
<button
  class="css-pmc161 css-2f61t6 css-1dhsa7o css-9fbhz8"
  data-testid="button-under-test"
>
  I am a 
  large
   
  default
   button
</button>
`;

exports[`<Button /> variants should render a \`primary\` <Button /> component 1`] = `
<button
  class="css-pmc161 css-2f61t6 css-13qzre css-zm3qx2"
  data-testid="button-under-test"
>
  I am a 
  small
   
  primary
   button
</button>
`;

exports[`<Button /> variants should render a \`secondary\` <Button /> component 1`] = `
<button
  class="css-pmc161 css-2f61t6 css-h1txb1 css-5cm1aq"
  data-testid="button-under-test"
>
  I am a 
  small
   
  secondary
   button
</button>
`;

exports[`<Button /> variants should render a default <Button /> component 1`] = `
<button
  class="css-pmc161 css-2f61t6 css-1dhsa7o css-9fbhz8"
  data-testid="button-under-test"
>
  I am a 
  small
   
  default
   button
</button>
`;
